client/http-client/src/client.rs

```diff
@@ -79,6 +80,17 @@ pub struct HttpClientBuilder {
 	id_kind: IdKind,
 	max_log_length: u32,
 	headers: HeaderMap,
+	proxy: Option<Uri>,
+}
+
+/// Error that may occur when trying to parse a `proxy URL`.
+#[derive(thiserror::Error, Debug)]
+struct InvalidProxyUrl(String);
```

Comment:
It looks like this is returned in an `Error`, so I think it should be `pub` so that it can be named by people.

Reply:
(and probably it should also be moved to live alongside `Error`? It feels a bit out of place here)

Reply:
Ah I see now' it's converted into an `anyhow::Error` before being returned; ignore me then :)